### ISSUE TITLE
Cannot install on Bazzite on Rog Ally

### ISSUE THREAD

**Issue by a94marbo** [jlobue10/rEFInd_GUI#3]
Install script seems to work.
Using the install with GUI:

Installing from Sourceforge: It seems to exit with no errors, but create config and Install config seem to be doing nothing.

Installing from Pacman: Installs exists with an error
"error: Operation aborted
sudo: refind-install: command not found
Invalid bootnum value: FFFFFFFFFFFFFFF5"

**Comment by jlobue10** [comment]
Please show me the output of `efibootmgr` and `sudo nano /boot/efi/efi/refind/refind.conf` . Install config should be working, without `sudo` permission password but I do need to update the visual feedback for that. Please consider this repo a work in progress variant of my SteamDeck_rEFInd repo (for more generic Linux uses).

**Comment by jlobue10** [comment]
Normally when rEFInd gets installed, it assumes the top boot priority, if the script worked successfully. One of the extra steps necessary on the Steam Deck is to disable the Windows EFI entry from taking highest priority away from rEFInd. Looks like that was successfully disabled here for you (no asterisk nest to the Windows entry). Can you try `sudo efibootmgr -o 0001,0000,0004` from the Bazzite command line? That should have rEFInd taking over the top boot priority.

**Comment by a94marbo** [comment]
nope, after reboot. no success

```
bazzite@bazzite:~$ sudo efibootmgr -v
[sudo] password for bazzite: 
BootCurrent: 0000
Timeout: 1 seconds
BootOrder: 0001,0000,0004
Boot0000* bazzite	HD(1,GPT,59242e08-1b74-41c3-92ef-8186a1c6880b,0x800,0x82000)/\EFI\fedora\shimx64.efi
      dp: 04 01 2a 00 01 00 00 00 00 08 00 00 00 00 00 00 00 20 08 00 00 00 00 00 08 2e 24 59 74 1b c3 41 92 ef 81 86 a1 c6 88 0b 02 02 / 04 04 34 00 5c 00 45 00 46 00 49 00 5c 00 66 00 65 00 64 00 6f 00 72 00 61 00 5c 00 73 00 68 00 69 00 6d 00 78 00 36 00 34 00 2e 00 65 00 66 00 69 00 00 00 / 7f ff 04 00
Boot0001* rEFInd	HD(1,GPT,59242e08-1b74-41c3-92ef-8186a1c6880b,0x800,0x82000)/\EFI\refind\refind_x64.efi
      dp: 04 01 2a 00 01 00 00 00 00 08 00 00 00 00 00 00 00 20 08 00 00 00 00 00 08 2e 24 59 74 1b c3 41 92 ef 81 86 a1 c6 88 0b 02 02 / 04 04 3a 00 5c 00 45 00 46 00 49 00 5c 00 72 00 65 00 66 00 69 00 6e 00 64 00 5c 00 72 00 65 00 66 00 69 00 6e 00 64 00 5f 00 78 00 36 00 34 00 2e 00 65 00 66 00 69 00 00 00 / 7f ff 04 00
Boot0004  Windows Boot Manager	HD(1,GPT,59242e08-1b74-41c3-92ef-8186a1c6880b,0x800,0x82000)/\EFI\Microsoft\Boot\bootmgfw.efi57494e444f5753000100000088000000780000004200430044004f0042004a004500430054003d007b00390064006500610038003600320063002d0035006300640064002d0034006500370030002d0061006300630031002d006600330032006200330034003400640034003700390035007d00000000000100000010000000040000007fff0400
      dp: 04 01 2a 00 01 00 00 00 00 08 00 00 00 00 00 00 00 20 08 00 00 00 00 00 08 2e 24 59 74 1b c3 41 92 ef 81 86 a1 c6 88 0b 02 02 / 04 04 46 00 5c 00 45 00 46 00 49 00 5c 00 4d 00 69 00 63 00 72 00 6f 00 73 00 6f 00 66 00 74 00 5c 00 42 00 6f 00 6f 00 74 00 5c 00 62 00 6f 00 6f 00 74 00 6d 00 67 00 66 00 77 00 2e 00 65 00 66 00 69 00 00 00 / 7f ff 04 00
    data: 57 49 4e 44 4f 57 53 00 01 00 00 00 88 00 00 00 78 00 00 00 42 00 43 00 44 00 4f 00 42 00 4a 00 45 00 43 00 54 00 3d 00 7b 00 39 00 64 00 65 00 61 00 38 00 36 00 32 00 63 00 2d 00 35 00 63 00 64 00 64 00 2d 00 34 00 65 00 37 00 30 00 2d 00 61 00 63 00 63 00 31 00 2d 00 66 00 33 00 32 00 62 00 33 00 34 00 34 00 64 00 34 00 37 00 39 00 35 00 7d 00 00 00 00 00 01 00 00 00 10 00 00 00 04 00 00 00 7f ff 04 00
```

**Comment by a94marbo** [comment]
well the conf file exist after I realized I could look with sudo tree /boot. But there is no efi file in the refind folder.

**Comment by a94marbo** [comment]
hmm, the install file is not there. How do I pull the correct one from github?

**Comment by jlobue10** [comment]
Oh yes, I had assumed that you tried to run [this script](https://github.com/jlobue10/rEFInd_GUI/blob/main/refind_install_Sourceforge.sh) . You can try just running this script and then redoing create config followed by install config. If that still doesn't work, then you'd need to actually run the `refind-install` command manually as I pointed out (and then lastly create config, install config afterwards). I do intend to clean this up at some point. Sorry for any inconvenience/ difficulty getting this up and running.

**Comment by jlobue10** [comment]
One other quality of life thing that you may want to apply to hide grub (after selecting the Bazzite boot option) is to run this command from cli.

```
ujust configure-grub hide
```

**Comment by a94marbo** [comment]
Ok, thanks. I got it working now. It was the localization issue that messed with me. The Downloads folder was in the wrong place.

Now secure boot wont work : ), but once I turned secure boot off, and fixed a new pin to Windows, everything seems to work without secure boot.
To enroll the hash of refind in MOK seems to be a science on its own....

**Comment by jlobue10** [comment]
Yes secure boot enabled with rEFInd is a bit more complicated. I have to refresh my memory about how exactly I got it to work on my ALLY and ALLY X (running Nobara). I'll respond to this thread more after work today, as available. In the meantime, please check available rEFInd documentation on this topic.

**Comment by a94marbo** [comment]
no worries, thanks alot!

**Comment by jlobue10** [comment]
For secure boot support, I followed the instructions [here](https://www.rodsbooks.com/refind/secureboot.html) installing one of the refind certs using mokutil. I then am using sbctl to sign all of my kernels in Nobara. I'd have to search my bash history on my ROG ALLY X if you need more info than that, as I can't remember all of the exact details.
